src/datachain/lib/listing.py

```diff
@@ -90,6 +91,10 @@ def _isfile(client: "Client", path: str) -> bool:
     Returns True if uri points to a file
```

Comment:
@shcheklein, was `_isfile()` added just for making it work with Google Cloud?

If so, I'd recommend extending `GCSFileSystem._is_file` (my preference), or add `GCSClient._is_file`.

Whitelisting exceptions on a fs-agnostic/generic code is not a good thing to do.

Reply:
no, I think it is a general code (there are special zero-byte files across all clouds I think) - we need to filter them

Reply:
Btw, we stopped doing these checks in dvc. We ignore files that end with slashes.

Reply:
yes, so that's the point of this call - to always treat it as a directory AFAIR

Reply:
I guess in some cases (not sure if this works the same across all clients) it returns a `file/` even if ask it to check `file` if zero-byte object exists

Reply:
I think this warrants separate discussion if we want to include _isfile or not. Merging the PR for now.

Reply:
The discussion is not limited to only `_isfile`.
I am arguing that `_is_file` should not raise `ClientError`, or handle certain set of exceptions. `_is_file` should raise any exception that is not `FileNotFoundError`, and `get_listing` should convert it to `ClientError` without any filtering. There's no need for whitelisting.

Reply:
`_isfile` should be very close to `fs._is_file`.

Reply:
I guess in this case we can consider is_file more like `client.is_file` that already abstracts out `fs` - thus it is probably fine to convert exceptions if that is relevant